### PR TITLE
Application service can join remote federated room without profile set

### PR DESCRIPTION
**Pull request by MadLittleMods**
Application service can join remote federated room without profile set

Synapse changes: https://github.com/matrix-org/synapse/pull/13131 to address https://github.com/matrix-org/synapse/issues/4778

### CODE REVIEW

File: tests/federation_room_join_test.go [hunk 1]
```diff
@@ -522,3 +522,28 @@ func TestSendJoinPartialStateResponse(t *testing.T) {
...
+	// Create the application service bridge user that is able to import historical messages
+	asUserID := "@the-bridge-user:hs1"
+	as := deployment.Client(t, "hs1", asUserID)
+
+	// Create the federated user which will fetch the messages from a remote homeserver
+	remoteUserID := "@charlie:hs2"
+	remoteCharlie := deployment.Client(t, "hs2", remoteUserID)
+
+	t.Run("join remote federated room as application service user", func(t *testing.T) {
+		//t.Parallel()
```
Comment (MadLittleMods) [inline tests/federation_room_join_test.go:539]: I see that we have `t.Parallel()` commented out elsewhere in this file. Do we care about it also being here?

File: tests/federation_room_join_test.go [hunk 2]
```diff
@@ -522,3 +522,28 @@ func TestSendJoinPartialStateResponse(t *testing.T) {
...
+		// Create the room from a remote homeserver
+		roomID := remoteCharlie.CreateRoom(t, map[string]interface{}{
+			"preset": "public_chat",
+			"name":   "hs2 room",
+		})
+
+		// Join the AS to the remote federated room (without a profile set)
+		as.JoinRoom(t, roomID, []string{"hs2"})
```
[the remark this answers is not in the document]
Comment (MadLittleMods) [reply tests/federation_room_join_test.go:547]: cc @kegsay

**Comment by kegsay** [comment]
I thought AS support wasn't fully implemented yet in Complement?

File: tests/federation_room_join_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kegsay) [reply tests/federation_room_join_test.go:547]: Right, okay. Then yeah the Dendrite image won't read the registration files yet. Skip them on Dendrite please.

**Review by kegsay — APPROVED** [review]
Otherwise LGTM

**Pull request merged**